tests/Fixtures/Autoloaded/ArrayOneItem/does-not-mutate-raw-array.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace ArrayOneItem_RawArray;
+
+class Test
+{
+    public function getCollection(): array
+    {
+        return [];
```

Comment:
What is the reason for not mutating a raw array?

Reply:
Well, there are several reasons:

1. First of all, when you return a raw array (not a variable) from the method/function, you are likely using object-like arrays, not collections. For example

    ```php
    public function getDefaults(): array
    {
        return [
            'a' => 0,
            'b' => 1,
            'c' => 2,
        ];
    }
    ```

    Mutating such an array will produce the following diff, which is very unfriendly:

    ```diff
    public function getDefaults(): array
    {
    -        return [
    -            'a' => 0,
    -            'b' => 1,
    -            'c' => 2,
    -        ];
    +        return count([
    +            'a' => 0,
    +            'b' => 1,
    +            'c' => 2,
    +        ]) > 1 ? array_splice([
    +            'a' => 0,
    +            'b' => 1,
    +            'c' => 2,
    +        ], 0, 1, true) : [
    +            'a' => 0,
    +            'b' => 1,
    +            'c' => 2,
    +        ];
    }
    ```

    This is hard to read and understand.

2. For object-like arrays, we already have the following mutator:

    ```diff
    public function getDefaults(): array
    {
        return [
    -       'a' => 0,
    +       'a' > 0,
            'b' => 1,
            'c' => 2,
        ];
    }
    ```

    Thus, if you are killing such mutant by test, you will 99% kill the `ArrayOneItem` mutant since you are checking the keys of the array. Here we will have a "duplication".

Reply:
Makes sense, forgot about that last mutator.